### ISSUE TITLE
GH Action for CI should *not* run when PR is merged

### ISSUE THREAD

**Issue by thedavidprice** [redwoodjs/redwood#660]
Effectively getting a duplicate CI run on merge. See example:
https://github.com/redwoodjs/redwood/actions/runs/126490259

Possible this was added in case commit merged to `master` directly. Seems there's a better way to handle this.

**Comment by kimadeline** [comment]
From the "The branch is out-of-date" messages in PRs I assume that they are required to be synced with `master` before merging. If this setting is left as-is, when would you want to run the CI workflow, just PR updates?

However if this check is set to "loose", then a CI run on merge should be fine, although it might break 🙃 

🌱 Side note: Should CI runs be triggered for any PR update (comments, reviews, edits), or just for when new code is pushed to the branch?

**Comment by thedavidprice** [comment]
Here are the settings from `.github/workflows/build-eslint-jest.yaml`:

```yaml
on:
  push:
    branches: [master]
  pull_request:
    types: [opened, edited, synchronize, reopened]
```

There are some funky (super technical term) things that were going on originally including duplicate, simultaneous runs. And at first I had the `master` protected so that PRs were the only way to push, but that broke our release process. 🤦‍♂️

So these current settings:
- ironed out the hiccups
- trigger for the appropriate PR events
- and cover us in the case there's a force commit directly to `master`

I haven't looked at the docs recently, but I think what would be ideal is an either/or for `push: branches: [master]` to only trigger if the push is forced and _not_ via PR. Make sense?

**Comment by kimadeline** [comment]
> that broke our release process.

Mind sharing more deets? 👀  Is it similar to the problem people are facing [here](https://github.community/t/how-to-push-to-protected-branches-in-a-github-action/16101)?

Looking at the [webhook event payload details](https://developer.github.com/webhooks/event-payloads/#pull_request), I can only think of a hacky (another super technical term 😎 ) way of running the workflow only in these cases (did I understand them correctly?):
- when a PR is merged into `master` either on the PR branch or on `master` (but only once)
- when a commit is pushed to `master`

The hack would be to skip the jobs if `github.event.pull_request.merged == true`, which would make the CI workflow run only  on `master`. It would be a per-job condition though, I couldn't find a way to add that kind of check to the entire workflow 😞

**Comment by jtoar** [comment]
@thedavidprice could you confirm that this is still happening and/or is still something you want to fix? The fact that there's only three workflows in this popup makes me think that it's been fixed:

<img width="915" alt="image" src="https://user-images.githubusercontent.com/32992335/145742065-e9f613ac-9ec5-4447-a8c6-2dfa7c41917b.png">

**Comment by jtoar** [comment]
@thedavidprice I don't think this is still happening so I'll close this one out for now. I'll keep it in mind in though as we prioritize working on e2e.
